tests/x11/rstudio_server.pm

```diff
@@ -62,13 +62,17 @@ sub run() {
     # log out at last and close firefox
     assert_and_click("rstudio_server-sign-out");
     assert_screen("rstudio_server-login-username");
-    send_key('alt-f4');
 
-    # optionally click away the close all tabs window
-    assert_screen([qw(rstudio_server-firefox_quit-and-close-tabs generic-desktop)]);
-    if (match_has_tag('rstudio_server-firefox_quit-and-close-tabs')) {
-        click_lastmatch();
-        assert_screen('generic-desktop');
+    for my $i (1 .. 5) {
+        send_key('ctrl-w', wait_screen_change => 1);
```

Comment:
Instead of this with the wait screen I still suggest to use a multi-tag assert screen. Exit the loop as soon as you match the desktop or if the loop runs out